### PR TITLE
append a feature help to change document.title

### PR DESCRIPTION
**Pull request by sqchenxiyuan**
by add `title` to route config 

like 

```
const router = new VueRouter({
  routes: [
    { path: '/', title: 'home', component: Home },
  ]
})
```

When the current page is specified, the title will be changed

----

reason: I have seen a request in the past , but not successfully completed the merge, and there is no continuation of the merger, so I pull this merge.

### CODE REVIEW

File: src/history/base.js [hunk 1]
```diff
@@ -139,10 +139,17 @@ export class History {
...
     this.cb && this.cb(route)
+    this.updateTitle(route)
     this.router.afterHooks.forEach(hook => {
       hook && hook(route, prev)
     })
   }
+
+  updateTitle (route: Route) {
+    if (route.title !== '') {
+      window.document.title = route.title
```
Comment (znck) [inline src/history/base.js:150]: IMO this would cause problem with SSR.

Comment (sqchenxiyuan) [reply src/history/base.js:150]: my code is not good       

dose there need a judge for window ?  or how to solve  this?  can you help me?

Comment (znck) [reply src/history/base.js:150]: /ping @posva

**Comment by znck** [comment]
This should not be the responsibility of the router, as it can be used in multiple environments.

Also it could be done with in after hook.

**Pull request closed without merging**